jsapp/js/projects/projectsTable/colResize.tsx

```diff
@@ -0,0 +1,271 @@
+        isDraggingRef.current &&
+        // Type safety
+        draggingColumnRef.current !== null &&
+        dragStartXRef.current !== null &&
+        dragStartWidthRef.current !== null &&
+        // De-dupe
+        dragPrevXRef.current !== e.pageX // skip event if same x
+      ) {
+        const newWidth = clampedColumnWidth(
+          draggingColumnRef.current,
+          // Calculate desired width based on initial positions and current x
+          dragStartWidthRef.current + (e.pageX - dragStartXRef.current)
+        );
+        // Update state for re-render only if the width is new after clamp
+        if (newWidth !== dragPrevWidthRef.current) {
+          dispatch({
+            type: 'resize',
+            column: draggingColumnRef.current,
+            width: newWidth,
+          });
+        }
+        // Set variables for de-duping
+        dragPrevXRef.current = e.pageX; // to skip event early if same x
+        dragPrevWidthRef.current = newWidth; // to skip state update if same width
+        return;
+      }
+
+      // Mousedown
+      //  - Start the dragging interaction (if on a resize handle)
+      if (
+        e.type === 'mousedown' &&
+        e.button === 0 // Only on left (primary) mouse button
+      ) {
+        // Detect resize handle with [data-resize-fieldname={fieldname}]
+        const fieldname = (e.target as HTMLElement).dataset.resizeFieldname;
+        if (fieldname) {
+          setShouldRenderBodyCursor(true);
+          isDraggingRef.current = true;
+          draggingColumnRef.current = fieldname;
+          dragStartXRef.current = e.pageX;
+          dragPrevXRef.current = e.pageX;
+
+          // Capture the current width of the resizer's parent element, a
+          // header cell, same width as all the row cells below it.
+          const parent = (e.target as HTMLElement).parentElement;
+          if (parent) {
+            dragStartWidthRef.current = parent.offsetWidth;
+            // box-sizing: border-box makes this very easy
+          }
+        }
+        return;
+      }
+
+      // Mouseup (or contextmenu)
+      //  - End the dragging interaction
+      //  - Clear the cursor
+      if (e.type === 'mouseup' || e.type === 'contextmenu') {
+        // // Uncomment to help find good width settings
+        // if (isDraggingRef.current) {
+        //   console.log(
+        //     'resized',
+        //     draggingColumnRef.current,
+        //     'to',
+        //     dragPrevWidthRef.current,
+        //     'px'
+        //   );
+        // }
+        isDraggingRef.current = false;
+        setShouldRenderBodyCursor(false);
+        return;
+      }
+    }
+  });
+  const eventTypes = ['mousedown', 'mouseup', 'mousemove', 'contextmenu'];
+
+  // Mount/Unmount effect: Attach/remove handlers on the body
+  useEffect(() => {
+    // Mount! Add handlers on body
+    eventTypes.forEach((eventType) => {
+      document.body.addEventListener(eventType, handlerRef.current);
```

Comment:
When `ColumnResizer` is being rendered (i.e. user is displaying `#/projects/abc123` route) then every mouse move will trigger the callback. The callback will not do anything unless mouse is pressed of course, but perhaps we could change this. What if we only add handlers after `mousedown` (or even better: a `mousedown` on the resizer element), and remove them after `mouseup`/`contextmenu`? This way the callback funcion wouldn't be called at all unless user is actually resizing.

Reply:
I might be too cautious with this thoug. Perhaps todays browsers handle these do-nothing functions with no performance loss. But still, this feels untidy :)

Reply:
It does seem like the right thing to do :) I combined them all to make it easier to read/write/test but I think it's turned out simple enough.

Reply:
I moved just the move handler out of this function and toggle it separately. I'm keeping the other handlers combined to simplify the code.